components/telemetry-operator/internal/kubernetes/configmap.go

```diff
@@ -0,0 +1,33 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ConfigmapProber struct {
+	client.Client
+}
+
+const overrideConfigFileName = "override-config"
+
+func (cmp *ConfigmapProber) ReadConfigMapIfPresent(ctx context.Context, name types.NamespacedName) (string, error) {
```

Comment:
The behaviour of returning an empty string if the CM does not exist might be not expected for that method name. Maybe let's name it `ReadConfigMapOrEmpty`